graylog2-server/src/main/java/org/graylog2/migrations/V20170110150100_FixAlertConditionsMigration.java

```diff
@@ -0,0 +1,150 @@
+    private static final String FIELD_ALERT_CONDITIONS_ID = "alert_conditions.id";
+    private static final String ALERT_CONDITIONS_PARAMETERS_PREFIX = "alert_conditions.$.parameters.";
+
+    private final MongoCollection<Document> collection;
+    private final ClusterConfigServiceImpl clusterConfigService;
+
+    @Inject
+    public V20170110150100_FixAlertConditionsMigration(final MongoConnection mongoConnection,
+                                                       final ClusterConfigServiceImpl clusterConfigService) {
+        this.collection = mongoConnection.getMongoDatabase().getCollection("streams");
+        this.clusterConfigService = clusterConfigService;
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.parse("2017-01-10T15:01:00Z");
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void upgrade() {
+        if (clusterConfigService.get(MigrationCompleted.class) != null) {
+            LOG.debug("Migration already done.");
+            return;
+        }
+
+        final ImmutableSet.Builder<String> modifiedStreams = ImmutableSet.builder();
+        final ImmutableSet.Builder<String> modifiedAlertConditions = ImmutableSet.builder();
+
+        for (Document document : collection.find().sort(ascending(FIELD_CREATED_AT))) {
+
+            final String streamId = document.getObjectId(FIELD_ID).toHexString();
+
+            if (!document.containsKey(FIELD_ALERT_CONDITIONS)) {
+                continue;
+            }
+
+            final List<Document> alertConditions = (List<Document>) document.get(FIELD_ALERT_CONDITIONS);
+
+            // Need to check if the following fields are integers:
+            //
+            // FieldContentValue: grace, backlog
+            // FieldValue:        grace, backlog, time, threshold
+            // MessageCount:      grace, backlog, time, threshold
+            final Set<String> intFields = ImmutableSet.of("grace", "backlog", "time", "threshold");
+
+            for (Document alertCondition : alertConditions) {
+                final String alertConditionId = alertCondition.get("id", String.class);
+                final Document parameters = alertCondition.get("parameters", Document.class);
+
+                for (String field : intFields) {
+                    final Object fieldValue = parameters.get(field);
+
+                    // No need to convert anything if the field does not exist or is already an integer
+                    if (fieldValue == null || fieldValue instanceof Integer) {
+                        continue;
+                    }
+
+                    if (!(fieldValue instanceof String)) {
+                        LOG.warn("Field <{}> in alert condition <{}> of stream <{}> is not a string but a <{}>, not trying to convert it!",
+                                field, alertConditionId, streamId, fieldValue.getClass().getCanonicalName());
+                        return;
```

Comment:
I think in this case we should only log the situation and continue converting other fields. This situation may happen with alert conditions created before 2.0. See https://github.com/Graylog2/graylog2-server/commit/ea4aa95fb609b56dcde4adc56cf311edf4ff3c0e for more information.